Jenkinsfile.windows

```
@@ -0,0 +1,281 @@
+@Library('pipeline-shared-libraries@feature/windows-pbp') _
+def utils
+
+pipeline {
+
+  agent { label "docker" }
+
+  options {
+    timestamps()
+    disableConcurrentBuilds()
+    buildDiscarder(
+      logRotator(
+        artifactDaysToKeepStr: '',
+        artifactNumToKeepStr: '',
+        daysToKeepStr: '',
+        numToKeepStr: '100'
+      )
+    )
+  }
+
+  parameters {
+    string(name: 'SLACK_CHANNEL', defaultValue: '#ide-builds', description: 'Slack channel to publish build message.')
+    string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
+    booleanParam(name: 'RSTUDIO_SKIP_QT', defaultValue: false, description: 'Skips installing and bulding for QT')
+    booleanParam(name: 'DAILY', defaultValue: false, description: 'Runs daily build if true')
+    booleanParam(name: 'PUBLISH', defaultValue: true, description: 'Runs publish stage if true')
+    gitParameter defaultValue: "${env.GIT_BRANCH}", name: 'COMMIT_HASH', type: 'PT_REVISION'
+  }
+
+  environment {
+    PACKAGE_OS = 'Windows'
+    AWS_ACCOUNT_ID = '749683154838'
+  }
+
+  stages {
+
+    stage('Load Utils') {
+      steps {
+        script {
+          sh 'printenv'
+          sh "echo 'Loading utils from ${env.WORKSPACE}/utils.groovy'"
+          utils = load "${env.WORKSPACE}/utils.groovy"
+        }
+      }
+    }
+
+    stage('Versioning') {
+      steps {
+        script {
+          (RSTUDIO_VERSION,
+            RSTUDIO_VERSION_MAJOR,
+            RSTUDIO_VERSION_MINOR,
+            RSTUDIO_VERSION_PATCH,
+            RSTUDIO_VERSION_SUFFIX) = utils.getVersion()
+          RSTUDIO_RELEASE = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}${RSTUDIO_VERSION_SUFFIX}"
+          RSTUDIO_FLOWER = readFile(file: 'version/RELEASE').replaceAll(" ", "-").toLowerCase().trim()
+          currentBuild.displayName = "${RSTUDIO_RELEASE}"
+          IS_PRO = RSTUDIO_VERSION_SUFFIX.contains('pro')
+        }
+      }
+    }
+
+    stage('Use Windows Agent') {
+      agent { label "windows" }
+
+      stages {
+        stage('Prepare Windows Container') {
+          stages {
+            stage('Checkout') {
+              when { expression { params.COMMIT_HASH !=  env.GIT_BRANCH } }
+              steps {
+                echo "Commit_hash value: ${params.COMMIT_HASH}"
+                checkout([$class: 'GitSCM',
+                   branches: [[name: "${params.COMMIT_HASH}"]],
+                   extensions: [],
+                   userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: 'https://github.com/rstudio/rstudio']]])
```

Sorry I missed this before, but we can use `${GIT_URL}` here instead, that way we won't have to change this in the pro repo.
```suggestion
                   userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: "${GIT_URL}"]]])
```

---

Does this also need a definition somewhere else, or do we already GIT_URL for free inside Jenkins?

---

Also could it just be 

```suggestion
                   userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: GIT_URL]]])
```

(no interpolation) since we're directly making use of an environment variable?

---

Yeah, I think you're right, we can just pass it directly. 

$GIT_URL will be predefined as an environment variable, so we don't need to set it anywhere.